apps/networkmonitor/networkmonitor.nim

```diff
@@ -128,6 +130,16 @@ proc setConnectedPeersMetrics(
 
   var analyzeFuts: seq[Future[Result[string, string]]]
 
+  var (inConns, outConns) = node.peer_manager.connectedPeers(WakuRelayCodec)
+  info "connected peers", inConns=inConns.len, outConns=outConns.len
+
+  shuffle(outConns)
+
+  if outConns.len >= toInt(MaxConnectedPeers/2):
+    for p in outConns[0 ..< toInt(outConns.len/2)]:
+      trace "Pruning Peer", Peer = $p
+      asyncSpawn(node.switch.disconnect(p))
```

Comment:
Out of curiosity, why disconnect only `out` and not `in` too?

Reply:
No particular reason, it is mostly NM discovering and connecting to other nodes, so the `out` is more important, but it is a good point to also make sure `in` is not filling up too much